appcues/src/main/java/com/appcues/analytics/ActivityRequestBuilder.kt

```diff
@@ -18,25 +19,27 @@ internal class ActivityRequestBuilder(
         const val SCREEN_TITLE_CONTEXT = "screen_title"
     }
 
-    fun identify(properties: Map<String, Any>? = null) = decorator.decorateIdentify(
+    fun identify(sessionId: UUID, properties: Map<String, Any>? = null) = decorator.decorateIdentify(
```

Comment:
would it work for `ActivityRequestBuilder` to have `SessionMonitor` injected? then it could just get `sessionId` directly rather than having to pass it in at all callsites?

The network object `ActivityRequest` could have this value optional, since technically that value in `SessionMonitor` is optional - but we know at this point it will always be set, due to the other checks that run before activity is tracked.

Reply:
since all the ActivityRequest should have sessionId I think its better if we enforce it with a non-null UUID.

assuming that it means that we should be getting sessionId from a place were we know its never null (AnalyticsTracker), even though this is more verbose and account for more change it explicitly say that session id is a non null required variable we need to pass on, we need to have a session in order to call the activity endpoint